cpp/daal/src/algorithms/k_nearest_neighbors/kdtree_knn_classification_training_input.cpp

```diff
@@ -0,0 +1,97 @@
+/* file: kdtree_knn_classification_training_input.cpp */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Implementation of K-Nearest Neighbors (kNN) algorithm classes.
+//--
+*/
+
+#include "algorithms/k_nearest_neighbors/kdtree_knn_classification_training_types.h"
+#include "src/services/serialization_utils.h"
+#include "src/services/daal_strings.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace kdtree_knn_classification
+{
+namespace training
+{
+namespace interface1
+{
+using namespace daal::data_management;
+using namespace daal::services;
+
+services::Status Input::check(const daal::algorithms::Parameter * parameter, int method) const
+{
+    return checkImpl(parameter);
+}
+
+services::Status Input::checkImpl(const daal::algorithms::Parameter * parameter) const
```

Comment:
is it possible to have some common function for similar check in base class https://github.com/oneapi-src/oneDAL/blob/master/cpp/daal/src/algorithms/classifier/classifier_train.cpp#L92 ?

Reply:
It's possible, but no reason to do it now due to new interfaces.